### PR TITLE
overridable univ3 fees/ticks literals for univ3 based dexes

### PR DESCRIPTION
**Pull request by rouzwelt**
## Motivation
- univ3 with overridable fees (for univ3 dexes with different fee literals than default univ3 such as pancakeswap v3).
- part of https://github.com/rainlanguage/arb-bot/issues/136

## Solution
- make uniswapv3 base class's fees and ticks overridable for child classes (dex classes)
- update `DataFetcher` tests to include checks for valid rp4 routes to ensures validity of dex literals
- add tests for reading fee/tick map from univ3 based dex factories and check against the literals in dexes

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs

### CODE REVIEW

File: packages/sushi/src/router/liquidity-providers/UniswapV3Base.ts [hunk 1]
```diff
@@ -435,25 +452,23 @@ export abstract class UniswapV3BaseProvider extends LiquidityProvider {
...
-          [tokenA, tokenB, SushiSwapV3FeeAmount.LOW],
-          [tokenA, tokenB, SushiSwapV3FeeAmount.MEDIUM],
-          [tokenA, tokenB, SushiSwapV3FeeAmount.HIGH],
-        ])
-      }
-      return []
-    }, [])
-
-    const filtered: [Token, Token, SushiSwapV3FeeAmount][] = []
+    const allCurrencyCombinationsWithAllFees: [Type, Type, number][] =
```
Comment (hardyjosh) [inline packages/sushi/src/router/liquidity-providers/UniswapV3Base.ts:455]: @rouzwelt can you pls match the formatting so this whole block doesn't show as changed?

**Review by hardyjosh — CHANGES_REQUESTED** [review]
I can't see tests for the changes that were made.

File: packages/sushi/src/router/liquidity-providers/UniswapV3Base.ts (hunk 1 above)
Comment (rouzwelt) [reply packages/sushi/src/router/liquidity-providers/UniswapV3Base.ts:455]: this is the repo linter setup, it lints like this automatically, if I change it then the lint tests will fail

**Comment by rouzwelt** [comment]
> I can't see tests for the changes that were made.

there was a missing commit, it is pushed now, but tests for dexes are in data fetcher where the loop test exists, running the datafetchere on each chain and dex and checking the results

**Review by hardyjosh — APPROVED** [review]

**Pull request merged**